#! /bin/sh /usr/share/dpatch/dpatch-run
## 50_fix_error_recovery_path_for_volume_switch_write.dpatch by Daniel T Chen <crimsun@ubuntu.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Backport fix from 1.0.14 to fix error recovery path for volume/switch
## DP: write.
## DP: http://hg-mirror.alsa-project.org/alsa-lib?cs=b58f3810a957;style=raw

@DPATCH@
diff -urNad alsa-lib-1.0.13~/src/mixer/simple_none.c alsa-lib-1.0.13/src/mixer/simple_none.c
--- alsa-lib-1.0.13~/src/mixer/simple_none.c	2006-09-29 07:42:56.000000000 -0400
+++ alsa-lib-1.0.13/src/mixer/simple_none.c	2007-02-04 17:49:09.000000000 -0500
@@ -515,7 +515,7 @@
 	return 0;
 }
 
-static int selem_write(snd_mixer_elem_t *elem)
+static int selem_write_main(snd_mixer_elem_t *elem)
 {
 	selem_none_t *s;
 	unsigned int idx;
@@ -598,6 +598,16 @@
 	return 0;
 }
 
+static int selem_write(snd_mixer_elem_t *elem)
+{
+	int err;
+	
+	err = selem_write_main(elem);
+	if (err < 0)
+		selem_read(elem);
+	return err;
+}
+
 static void selem_free(snd_mixer_elem_t *elem)
 {
 	selem_none_t *simple = snd_mixer_elem_get_private(elem);
